### PR TITLE
update umpire to match camp and raja exports

### PR DESCRIPTION
**Pull request by trws**
I'm trying this as a solution to the build failure in the upstream spack update: https://gitlab.spack.io/spack/spack/-/jobs/2875546

Thinking we need some more downstream testing too, especially with BLT-based projects.

### CODE REVIEW

**Comment by trws** [comment]
@davidbeckingsale, @adrienbernede, is there a way to prevent icpc 18 from breaking here without going to the radiuss shared ci stuff?

**Comment by davidbeckingsale** [comment]
Does Intel 18 work if you build as c++14 instead of 17? Seems like the it's working for RAJA so I would expect it should work here without major changes.

**Comment by trws** [comment]
It has been removed from raja already as part of this process, we're explicitly supporting 19 but not 18 for some time now was my understanding.  I don't know, is the direct answer to your question however.

**Comment by davidbeckingsale** [comment]
Oh, in that case, just ditch it here too :)

**Comment by trws** [comment]
So, I went to do that, but it's being pulled in from a separate RADIUSS repo AFAICT?

**Comment by davidbeckingsale** [comment]
Good point - that's the "shared" CI jobs. I've made a branch, should be able to point Umpire at it and check everything works. The `ref` for the shared include is here: https://github.com/LLNL/Umpire/blob/develop/.gitlab-ci.yml#L40 and you can change it to `task/remove-intel-18` (https://github.com/LLNL/radiuss-shared-ci/pull/2)

**Comment by trws** [comment]
Thanks for that @davidbeckingsale!  I was having trouble finding the target repo embarassingly enough, but the last set of tests got to all but a couple on XL thanks to that.  Hopefully with the bltTargets include protection we'll make it and be able to close this out.

**Comment by trws** [comment]
Looks like we're green here. @rhornung, @davidbeckingsale do you want to do other testing here or with the whole unit before we merge to umpire/raja?

**Comment by davidbeckingsale** [comment]
I'm good with this.

**Comment by trws** [comment]
Cool, then feel free to approve and/or merge as desired.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ stages:
     include:
       - local: '.gitlab/custom-jobs.yml'
       - project: 'radiuss/radiuss-shared-ci'
-        ref: v2022.06.0
+        ref: task/remove-intel-18
```
Comment (adrienbernede) [inline .gitlab-ci.yml]: Let’s merge `task/remove-intel-18`, and maybe tag the merge `v2022.06.1` before we merge things here.
@trws I know it seems like running down a rabbit hole, am I thankful for your hard work.

File: cmake/SetupUmpireThirdParty.cmake [hunk 2]
```diff
@@ -85,7 +85,24 @@ if (UMPIRE_ENABLE_MPI OR UMPIRE_ENABLE_HIP OR UMPIRE_ENABLE_OPENMP OR UMPIRE_ENA
...
+  # If the target is EXPORTABLE, add it to the export set
+  get_target_property(_is_imported mpi IMPORTED)
+  if(NOT ${_is_imported})
+    install(TARGETS              mpi
+      EXPORT               ${arg_EXPORT})
+    # Namespace target to avoid conflicts
+    set_target_properties(mpi PROPERTIES EXPORT_NAME blt::mpi)
+    install(EXPORT bltTargets
+      DESTINATION  lib/cmake/umpire)
+  endif()
```
Comment (adrienbernede) [inline cmake/SetupUmpireThirdParty.cmake]: Wouldn’t this be better in the the scope of the previous `if`, i.e. only when BLT tpls are needed ?

File: .gitlab-ci.yml (hunk 1 above)
Comment (adrienbernede) [reply .gitlab-ci.yml]: Done

File: cmake/SetupUmpireThirdParty.cmake (hunk 2 above)
Comment (trws) [reply cmake/SetupUmpireThirdParty.cmake]: yes, probably.

Comment (adrienbernede) [reply cmake/SetupUmpireThirdParty.cmake]: OK, done.

File: .gitlab-ci.yml (hunk 1 above)
Comment (trws) [inline .gitlab-ci.yml]:
```suggestion
        ref: v2022.06.1
```

Comment (adrienbernede) [reply .gitlab-ci.yml]: I already applied the required change.

Comment (trws) [inline .gitlab-ci.yml]: Yup, saw after I did that, planning to remove this in a sec.  Thanks BTW!

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**